requirements/updater.sh

```diff
@@ -1,62 +1,69 @@
 #!/usr/bin/env bash
+# -*- coding: utf-8; mode: sh; sh-indentation: 2; indent-tabs-mode: nil; sh-basic-offset: 2; -*-
+
 set -ue
 
 PYTHON=python3.11
 
 for FILE in requirements.in requirements_all.txt ; do
-	if [ ! -f ${FILE} ] ; then
-		touch ${FILE}
-	fi
+  if [ ! -f ${FILE} ] ; then
+    touch ${FILE}
+  fi
 done
 requirements_dir="$(readlink -f .)"
 requirements_txt="$(readlink -f ./requirements_all.txt)"
 pip_compile="pip-compile --no-header --quiet -r --allow-unsafe"
 
 _cleanup() {
-  cd /
   test "${KEEP_TMP:-0}" = 1 || rm -rf "${_tmp}"
 }
 
 generate_requirements() {
-  venv="`pwd`/venv"
-  echo $venv
+  venv="$(pwd)/venv"
+  echo "Using virtual environment: $venv"
   ${PYTHON} -m venv "${venv}"
   # shellcheck disable=SC1090
   source ${venv}/bin/activate
 
-  ${venv}/bin/python -m pip install -U 'pip' pip-tools
+  ${venv}/bin/pip install -U pip pip-tools
+
+  ${pip_compile} ${requirements_dir}/requirements*.in --output-file requirements_all.txt
 
-  ${pip_compile} $(ls ${requirements_dir}/requirements*.in | xargs) --output-file requirements_all.txt
+  for file in ${requirements_dir}/requirements*.in; do
+    app="$(basename $file .in)"
+    echo "Compiling deps for $app"
+    ${pip_compile} $file --output-file "$app.txt" -c requirements_all.txt
```

Comment:
-c provides consistency across _all_ the deps, rather than resolving the deps for each app independently. Otherwise you end up with some apps trying to pull in Django 5 for example.